helix-rest/src/main/java/org/apache/helix/rest/metadatastore/accessor/ZkRoutingDataWriter.java

```diff
@@ -89,15 +94,27 @@ public ZkRoutingDataWriter(String namespace, String zkAddress) {
     String hostName = System.getProperty(MetadataStoreRoutingConstants.MSDS_SERVER_HOSTNAME_KEY);
     if (hostName == null || hostName.isEmpty()) {
       throw new IllegalStateException(
-          "Unable to get the hostname of this server instance. System.getProperty fails to fetch "
+          "Unable to get the hostname of this server instance or the hostname is empty. System.getProperty fails to fetch "
```

Comment:
"Host name is not set or is empty" would be better.

Reply:
Ok.